### PR TITLE
Update our workflows to the new latest docker builds as well fix nRF builds

### LINKED ISSUE #4655: nRF Docker image doesn't build
#### Problem
```
 ---> Running in 7f43ad14d5bf
+ mkdir -p /opt/NordicSemiconductor/nRF5_tools
+ cd /opt/NordicSemiconductor/nRF5_tools
+ curl https://www.nordicsemi.com/-/media/Software-and-other-downloads/Desktop-software/nRF-command-line-tools/sw/Versions-10-x-x/10-7-0/nRFCommandLineTools1070Linuxamd64tar.gz
+ tar zxvf -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   121  100   121    0     0     83      0  0:00:01  0:00:01 --:--:--    83

gzip: stdin: not in gzip format
tar: Child returned status 1
tar: Error is not recoverable: exiting now
The command '/bin/sh -c set -x     && (mkdir -p /opt/NordicSemiconductor/nRF5_tools && cd /opt/NordicSemiconductor/nRF5_tools     && curl https://www.nordicsemi.com/-/media/Software-and-other-downloads/Desktop-software/nRF-command-line-tools/sw/Versions-10-x-x/10-7-0/nRFCommandLineTools1070Linuxamd64tar.gz     | tar zxvf -     && dpkg -i --force-depends JLink_Linux_*.deb     && dpkg -i --force-depends nRF-Command-Line-Tools_*.deb     && tar zxvf nRF-Command-Line-Tools_*.tar.gz)     && rm -rf /opt/NordicSemiconductor/nRF5_tools/*.tar.gz     && rm -rf /opt/NordicSemiconductor/nRF5_tools/*.deb     && : # last line' returned a non-zero code: 2
```
 #### Proposed Solution
Fix

### PR DESCRIPTION
**Pull request by woody-apple**
#### Problem
Our docker builds are... a few builds back, and when I went to update things - some stuff seems to not be working.

 #### Summary of Changes
* Updated the nRF paths, Fixes: #4655 
* Updated the docker builds to latest (for now, working on proper Docker build image CI)
* Rebuilt all docker images based with no changes (except nRF paths) - packages would be updated, but nothing else

#### Help needed
* For some reason **some** DNS tests are failing, though **all** accessory builds work just fine, including QEMU
* Main builds fail (same as above) - weird, we seem to be running the same jobs twice...
* Cirque tests fail (possible same as above)

### CODE REVIEW

**Comment by woody-apple** [comment]
@andy31415 Could you take a look at why the DNS tests are failing?

**Comment by woody-apple** [comment]
> @andy31415 Could you take a look at why the DNS tests are failing?

Super weird. Fails in my fork, works here... 🤪

**Comment by woody-apple** [comment]
> @andy31415 Could you take a look at why the DNS tests are failing?
I also see what's up with cirque - will work on a temporary version strategy, then integrate this into our docker build pipeline.

**Comment by woody-apple** [comment]
@andy31415 @jelderton @saurabhst @bzbarsky-apple 

Ready for review (Note, will be following up about the "latest" tag in: https://github.com/project-chip/connectedhomeip/issues/3411)

Also filed: 
https://github.com/project-chip/connectedhomeip/issues/4658
https://github.com/project-chip/connectedhomeip/issues/4659

File: src/app/tests/integration/Dockerfile.initiator [hunk 1]
```diff
@@ -1,4 +1,21 @@
-from chip-cirque-device-base
+#
+#   Copyright (c) 2020 Project CHIP Authors
```
Comment (andy31415) [inline src/app/tests/integration/Dockerfile.initiator:2]: 2021

Is this license notice really required?

**Comment by andy31415** [comment]
One worry in using ':latest' is that it will couple docker releases with builds.

Currently if I build on version A and need a breaking change to build with B (e.g. a SDK update that requires code changes for compatibility), currently I can have:

- PR to update docker to B. Code uses A still.
- upload new docker (now both A and B are in docker)
- PR to update code to B and use image B.

Throughout I would have no CI breakages because code specifies with what image it is compatible.

If code uses ":latest" I am unsure how the label should be updated since as soon as I change the label, all code would use the updated image.

@woody-apple - thoughts? Are the benefits of not needing to keep updating versions better than the downside of not being able to push incompatible SDK versions (and some non-reproducible builds, since checking out old commits may still use new SDKs) ?

File: src/app/tests/integration/Dockerfile.initiator (hunk 1 above)
Comment (woody-apple) [reply src/app/tests/integration/Dockerfile.initiator:2]: I added it to others for consistency, it was in some, but not others.

**Comment by woody-apple** [comment]
> One worry in using ':latest' is that it will couple docker releases with builds.
> 
> Currently if I build on version A and need a breaking change to build with B (e.g. a SDK update that requires code changes for compatibility), currently I can have:
> 
> * PR to update docker to B. Code uses A still.
> * upload new docker (now both A and B are in docker)
> * PR to update code to B and use image B.
> 
> Throughout I would have no CI breakages because code specifies with what image it is compatible.
> 
> If code uses ":latest" I am unsure how the label should be updated since as soon as I change the label, all code would use the updated image.
> 
> @woody-apple - thoughts? Are the benefits of not needing to keep updating versions better than the downside of not being able to push incompatible SDK versions (and some non-reproducible builds, since checking out old commits may still use new SDKs) ?

As mentioned above, this is temporary - I'm working on our proper docker CI, and this will at least keep our builds more up to date (some of the old workflows were vastly out of date).

**Review by jelderton — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by woody-apple** [comment]
I hate to pile on, but this is resolved here too:
https://github.com/project-chip/connectedhomeip/issues/4670

**Comment by woody-apple** [comment]
@saurabhst ?

**Comment by woody-apple** [comment]
Merging due to hot-fix, due to broken build

**Pull request merged**